packages/apollo-datasource-rest/src/connectors/memcached.ts

```diff
@@ -60,11 +73,18 @@ export default class MemcachedKeyValueCache implements KeyValueCache {
 
   async invalidate(tags: string[]): Promise<void> {
     // set the invalidation "timestamp" using logical clock for every tag
-    const version = ++this.logicalClock;
+    let version = await this.client.get(VERSION_KEY);
```

Comment:
I wonder if we can do this in a more efficient way, because checking to see whether the key exists first on every increment seems expensive. Not sure if this is part of the underlying protocol and whether our client supports it, but it seems `libmemcached` has a [`memcached_increment_with_initial`](http://docs.libmemcached.org/memcached_auto.html) function.